crates/typst-syntax/src/package.rs

```diff
@@ -265,3 +430,73 @@ impl<'de> Deserialize<'de> for PackageVersion {
         string.parse().map_err(serde::de::Error::custom)
     }
 }
+
+/// A version bound for compatibility specification.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct SemVerBound {
```

Comment:
I'd prefer the name `VersionBound`, since our version type is also not strictly semver (less powerful).